src/libraries/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -2076,6 +2077,95 @@ public async Task SendHttpGetRequest_WithCachePolicyBypassCache_DoNotAddHeaders(
             });
         }
 
+        [Fact]
+        public async Task SendHttpPostRequest_WithContinueTimeoutAndBody_BodyIsDelayed()
+        {
+            await LoopbackServer.CreateClientAndServerAsync(
+                async (uri) =>
+                {
+                    HttpWebRequest request = WebRequest.CreateHttp(uri);
+                    request.Method = "POST";
+                    request.ServicePoint.Expect100Continue = true;
+                    request.ContinueTimeout = 30000;
+                    Stream requestStream = await request.GetRequestStreamAsync();
+                    requestStream.Write("aaaa\r\n\r\n"u8);
+                    var _ = request.GetResponseAsync();
+                },
+                async (server) =>
+                {
+                    await server.AcceptConnectionAsync(async (client) => 
+                    {
+                        await client.ReadRequestHeaderAsync();
+                        // This should time out, because we're expecting the body itself but we'll get it after 30 sec.
+                        await Assert.ThrowsAsync<TimeoutException>(() => client.ReadLineAsync().WaitAsync(TimeSpan.FromMilliseconds(100)));
+                    });
+                }
+            ).WaitAsync(TestHelper.PassingTestTimeout);
+        }
+
+        [Theory]
+        [InlineData(true, 1)]
+        [InlineData(false, 30000)]
+        public async Task SendHttpPostRequest_WithContinueTimeoutAndBody_Success(bool expect100Continue, int continueTimeout)
+        {
+            await LoopbackServer.CreateClientAndServerAsync(
+                async (uri) =>
+                {
+                    HttpWebRequest request = WebRequest.CreateHttp(uri);
+                    request.Method = "POST";
+                    request.ServicePoint.Expect100Continue = expect100Continue;
+                    request.ContinueTimeout = continueTimeout;
+                    Stream requestStream = await request.GetRequestStreamAsync();
+                    requestStream.Write("aaaa\r\n\r\n"u8);
+                    var _ = request.GetResponseAsync();
+                },
+                async (server) =>
+                {
+                    await server.AcceptConnectionAsync(async (client) => 
+                    {
+                        await client.ReadRequestHeaderAsync();
+                        // This should not time out, because we're expecting the body itself and we should get it after 1 sec.
+                        string data = await client.ReadLineAsync().WaitAsync(TimeSpan.FromSeconds(10));
+                        Assert.StartsWith("aaaa", data);
+                    });
+                }
+            ).WaitAsync(TestHelper.PassingTestTimeout);
```

Comment:
```suggestion
                });
```

Nit: When using `CreateClientAndServerAsync`, there is no need to set the timeout manually. The helper does that for you already.